src/main/java/ch/heigvd/dai/ios/AudioChangeVolume.java

```diff
@@ -1,9 +1,77 @@
 package ch.heigvd.dai.ios;
 
+import javax.sound.sampled.AudioFileFormat;
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import java.io.*;
+import java.nio.file.Paths;
+
 public class AudioChangeVolume implements VolumeModifiable {
 
+    private static final int BUFFER_SIZE = 1024; // Buffer size for reading audio data
+
+
     @Override
-    public void changeVolume(String filename, float intensity) {
-        throw new UnsupportedOperationException("Please remove this exception and implement this method.");
+    public void changeVolume(String filename, float volumeIntensity) {
+
+        try (InputStream audioSrc = new FileInputStream(filename);
+             InputStream bufferedAudioSrc = new BufferedInputStream(audioSrc);
+             AudioInputStream ais = AudioSystem.getAudioInputStream(bufferedAudioSrc);
+             ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+
+            AudioFormat format = ais.getFormat();
+
+            processAudio(ais, baos, volumeIntensity);
+
+            String newOutputFilename = getOutputFilename(filename, volumeIntensity);
+
+            byte[] adjustedAudio = baos.toByteArray();
+            File outputFile = new File(newOutputFilename);
+
+            try ( ByteArrayInputStream bais = new ByteArrayInputStream(adjustedAudio);
+                  AudioInputStream adjustedAudioStream = new AudioInputStream(bais, format, adjustedAudio.length / format.getFrameSize())) {
+
+
+                AudioSystem.write(adjustedAudioStream, AudioFileFormat.Type.WAVE, outputFile);
```

Comment:
This implies that only .wav files will be outputted. Is this wanted or would there be a possibility to output the same file type as the inputted file ?

Reply:
It was not intended to be so. Correction has been made.